### ISSUE TITLE
OpenSSL 

### ISSUE THREAD

**Issue by pojntfx** [open-iscsi/open-isns#20]
I tried to build `open-isns` using the following version:

```bash
[pojntfx@thinkpadx1c3 ~]$ openssl version
OpenSSL 1.1.1c FIPS  28 May 2019
```

Which fails due to these functions no longer being used: https://linux.die.net/man/3/openssl_add_all_digests.

Replacing

```c
OpenSSL_add_all_algorithms();
OpenSSL_add_all_ciphers();
OpenSSL_add_all_digests();
```
in `pki.c` with just

```c
OPENSSL_init_crypto();
```
fixes the issue.

**Comment by gonzoleeman** [comment]
I have created a new branch, under https://github.com/gonzoleeman/open-isns, branch handle-new-openssl. Note that you have to run "./configure --with-security ..." to build a Makefile that tests the code. But you probably already knew that, or you wouldn't have run into this problem! Surprisingly, the version I put in the SUSE open build service does not enable security, so I could never have seen this.

Side note: if you use open-isns, please send me a side note about how. I'm trying to expand it's use in general.

**Comment by gonzoleeman** [comment]
It case it was not clear, can you test that new branch and see if it fixes your issue?

**Comment by pojntfx** [comment]
@gonzoleeman Thanks for taking a look at this!

It appears to work now:

```bash
thinkpadx1c3% docker run -it ubuntu bash
root@8c5c7d67e558:/open-isns# history
    4  apt update
    5  apt install -y git build-essential
    7  git clone https://github.com/open-iscsi/open-isns.git
    8  cd open-isns/
   12  git checkout handle-new-openssl
   26  apt install libssl-dev
   27  ./configure --with-security
   28  make
```

I'm using `open-isns` and `open-iscsi` for OpenEBS on Kubernetes; I'm thinking about creating a distribution of `k3s` and `open-iscsi` to enable distributed storage, and I hit the issue above while doing some experiments.

**Comment by gonzoleeman** [comment]
Fixed in master, pull request #22
